backend/app/processing/db_storage.py

```
@@ -0,0 +1,57 @@
+    def save_uploaded_file(self, chatbot_id: str, file_obj, original_filename: str) -> Path:
+        db_dir = self.get_chatbot_dir(chatbot_id)
+        new_filename = f"{uuid.uuid4()}_{original_filename}"
+        dest_path = db_dir / new_filename
+        # Atomic write
+        tmp = dest_path.with_suffix(dest_path.suffix + ".tmp")
+        with open(tmp, "wb") as buf:
+            shutil.copyfileobj(file_obj, buf)
+        tmp.rename(dest_path)
+        return dest_path
+
+    def save_config(self, chatbot_id: str, config: Dict[str, Any]) -> Path:
```

Ruff (0.8.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)